io-engine/src/bin/io-engine-client/nexus_cli.rs

```diff
@@ -12,27 +12,56 @@ use mayastor_api::{v0, v1};
 use snafu::ResultExt;
 use tonic::{Code, Status};
 use uuid::Uuid;
+use mayastor_api::v1::nexus::ShareProtocol;
 
 pub fn subcommands<'a, 'b>() -> App<'a, 'b> {
     let create = SubCommand::with_name("create")
         .about("Create a new nexus device")
         .arg(
-            Arg::with_name("uuid")
+            Arg::with_name("name")
                 .required(true)
                 .index(1)
+                .help("name of the nexus"),
+        )
+        .arg(
+            Arg::with_name("uuid")
+                .required(true)
                 .help("uuid for the nexus, if uuid is not known please provide \"\" to autogenerate"),
         )
         .arg(
             Arg::with_name("size")
                 .required(true)
-                .index(2)
                 .help("size with optional unit suffix"),
         )
+        .arg(
+            Arg::with_name("min-cntlid")
+                .required(true)
+                .help("minimum NVMe controller ID for sharing over NVMf"),
+        )
+        .arg(
+            Arg::with_name("max-cntlid")
+                .required(true)
+                .help("maximum NVMe controller ID"),
+        )
+        .arg(
+            Arg::with_name("resv-key")
+                .required(true)
+                .help("NVMe reservation key for children"),
+        )
+        .arg(
+            Arg::with_name("preempt-key")
+                .required(true)
+                .help("NVMe preempt key for children, 0 for no preemption"),
+        )
+        .arg(
+            Arg::with_name("nexus-info-key")
+                .required(true)
+                .help("Key used to persist the NexusInfo structure to the persistent store"),
+        )
         .arg(
```

Comment:
CreateNexusRequest supports also reservation type (optional):

```
optional NvmeReservation resvType = 10; // the reservation type to use
```

which is one of the following possible values:

```
/// NVMe reservation types.
enum NvmeReservation {
  Reserved = 0;
  WriteExclusive = 1;
  ExclusiveAccess = 2;
  WriteExclusiveRegsOnly = 3;
  ExclusiveAccessRegsOnly = 4;
  WriteExclusiveAllRegs = 5;
  ExclusiveAccessAllRegs = 6;
}
```

None resolves to '''WriteExclusiveAllRegs'''. IMHO, for test purposes, it would be nice to support 'resvType' in CLI args too.

Reply:
Got it

Reply:
Incorporated